File: 1996/6/3-6.rb
#!/usr/bin/env ruby

TRASH = /([,.!?:;\n])\Z/
SWAP = /\A(.*)([TtKkWwSs]h|[Tt]r|[Ff]l)\Z/

File.readlines('pig.dat').each do |line|
  line.split.each do |word|
    word = word.gsub(TRASH, '')
    ending = $1.nil? ? '' : $1
    word = word[0...-2]
    if word =~ SWAP
      word = $2 + $1
    elsif word[-1,1] == 'y'
      word = word[0...-1]
    else
      word = word[-1,1] + word[0...-1]
    end
    print word + ending + ' '
  end
  puts
end

File: 1996/6/3-6.py
#!/usr/bin/env python

import sys

trash = [",", ".", "!", "?", "\n", ":", ";"]
swap = ["Th", "Kh", "Wh", "Tr", "Sh", "Fl", "th", "kh", "wh", "tr", "sh", "fl"]

for line in sys.stdin:
    if line == "":
        break
    words = line.split()
    for word in words:
        end = ""
        for trsh in trash:
            if word[-1] == trsh:
                word = word[:-1]
                end = trsh
                break
        word = word[:-2]
        done = 0
        for swp in swap:
            if word[-2:] == swp:
                word = swp + word[:-2]
                done = 1
                break
        if done == 0:
            if word[-1] == "y":
                word = word[:-1]
            else:
                word = word[-1] + word[:-1]
        print "%s%s" % (word, end),
